src/placeos/api_wrapper/drivers.cr

```diff
@@ -2,34 +2,17 @@ require "./endpoint"
 
 module PlaceOS
   class Client::APIWrapper::Drivers < Client::APIWrapper::Endpoint
+    include Client::APIWrapper::Endpoint::Search(Driver)
     include Client::APIWrapper::Endpoint::Fetch(Driver)
+    # include Client::APIWrapper::Endpoint::Create(Driver)
+    # include Client::APIWrapper::Endpoint::Update(Driver)
     include Client::APIWrapper::Endpoint::Destroy
 
     getter base : String = "#{API_ROOT}/drivers"
 
-    # List or search for drivers.
-    #
-    # Results maybe filtered by specifying a query - *q* - to search across driver
-    # attributes. A small query language is supported within this:
-    #
-    # Operator | Action
-    # -------- | ------
-    # `+`      | Matches both terms
-    # `|`      | Matches either terms
-    # `-`      | Negates a single token
-    # `"`      | Wraps tokens to form a phrase
-    # `(`  `)` | Provides precedence
-    # `~N`     | Specifies edit distance (fuzziness) after a word
-    # `~N`     | Specifies slop amount (deviation) after a phrase
-    #
-    # Up to *limit* drivers will be returned, with a paging based on *offset*.
-    def search(q : String? = nil, limit : Int = 20, offset : Int = 0)
-      get base, params: from_args, as: Array(Driver)
-    end
-
     def create(
       name : String,
-      role : Role,
+      role : Int32,
```

Comment:
would be good to have this using an explicit enum.

try making an alias to the actual enum where you generate the models